### PR TITLE
fix: update `text` variant of Button to align with design

### PR DESCRIPTION
**Pull request by jaredcwhite**
## Issue Overview

This PR addresses QA feedback as noted in #36 

## Description

The `borderless-text` variant has been renamed to `text`, and padding has been removed. There's also just small and medium sizes now (specifying a large size will look identical to medium).

## How Can This Be Tested/Reviewed?

Storybook: https://deploy-preview-58--storybook-ui-seeds.netlify.app/?path=/story/actions-button--text-buttons

## Checklist:

- [x] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have exported any new components
- [x] My commit message(s) and PR title are polished in the conventional commit format, and any breaking changes are indicated in the message and are well-described

### CODE REVIEW

File: src/actions/Button.scss [hunk 1]
```diff
@@ -135,13 +135,26 @@
...
+    border-radius: 0;
+    padding: 0;
     color: var(--seeds-color-primary);
     font-weight: normal;
     text-decoration: underline;
     gap: var(--seeds-s1);
+
+    &[data-size="sm"] {
+      padding: 0;
+      border-radius: 0;
```
Comment (ludtkemorgan) [inline src/actions/Button.scss:150]: These are the same values as above is this needed anymore? Same for the data-size lg below

**Review by ludtkemorgan — DISMISSED** [review]

File: src/actions/Button.scss [hunk 2]
```diff
@@ -135,13 +135,26 @@
...
+    }
+
+    &[data-size="lg"] {
+      padding: 0;
+      font-size: var(--button-font-size-md);
```
[the remark this answers is not in the document]
Comment (jaredcwhite) [reply src/actions/Button.scss:155]: @ludtkemorgan Yeah, the large size is "invalid" for the text variant, so we just make it look identical to medium (see Jesse's comments in original issue).

File: src/actions/Button.scss (hunk 1 above)
Comment (jaredcwhite) [reply src/actions/Button.scss:151]: Good point, some of the default vs. "md" size styles were hard to parse out…have made this clearer throughout.
